src/jira.servicedesk/client.bal

```diff
@@ -0,0 +1,366 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/auth;
+import ballerina/http;
+import ballerina/oauth2;
+
+# Handles global functions related to the `Jira Service Desk` instance.
+public type Client client object {
+    http:Client jiraClient;
+
+    # Initializes the Jira client endpoint.
+    #
+    # + jiraConfig - Jira client configuration record.
+    public function __init(Configuration jiraConfig) {
+        string baseUrl = jiraConfig.baseUrl + API_PATH;
+        http:ClientConfiguration clientConfig;
+        BasicAuthConfiguration|oauth2:DirectTokenConfig authConfig = jiraConfig.authConfig;
+        if (authConfig is BasicAuthConfiguration) {
+            auth:OutboundBasicAuthProvider outboundBasicAuthProvider = new ({
```

Comment:
Shall we rename for consistency with `oauth2Provider`?
```suggestion
            auth:OutboundBasicAuthProvider basicAuthProvider = new ({
```

Reply:
Fixed.